soccer/gameplay/skills/goalside_mark.py

```diff
@@ -0,0 +1,194 @@
+import single_robot_behavior
+import behavior
+import robocup
+import main
+import constants
+import role_assignment
+
+class Goalside_Mark(single_robot_behavior.SingleRobotBehavior):
+    '''
+    The goalside mark will take a robot or mark point and position itself on line with the best shot from 
+    the position of the robot/mark_point.
+    @param: ratio - this will control how far from the opposing robot the defender will mark (0 is close to robot)
+    '''
+
+    
+    def __init__(self):
+        super().__init__(continuous=True)
+        #Below params are described above @properties
+        self._ratio = 0.2
+        #This threshold is lower than mark to ensure we get goal side of the opponent fast, 
+        #then back off based on ratio
+        
+        self._mark_line_thresh = constants.Robot.Radius*3
+        self._mark_robot = None 
+        self._mark_point = None 
+
+        self._target_point = None 
+
+        self.add_transition(behavior.Behavior.State.start,
+                            behavior.Behavior.State.running, lambda: True,
+                            "immediately")
+        self.kick_eval = robocup.KickEvaluator(main.system_state())
+
+    def execute_running(self):
+        #pylint: disable=no-member
+        #Skill does nothing if mark point isn't given AND the ball or robot to mark can't be found
+        self.kick_eval = robocup.KickEvaluator(main.system_state())
```

Comment:
I don't think you need to create a new kick eval object every frame. You can just use the same one and clear the excluded robots whenever

Reply:
Good call- changed this trying to debug the kickevaluator and never changed it back